Assets/MixedRealityToolkit/Services/BaseDataProvider.cs

```diff
@@ -38,11 +40,16 @@ public abstract class BaseDataProvider : BaseService, IMixedRealityDataProvider
         /// <param name="priority">The registration priority of the data provider.</param>
         /// <param name="profile">The configuration profile for the data provider.</param>
         protected BaseDataProvider(
-            IMixedRealityService service,
+            T service,
             string name = null,
             uint priority = DefaultPriority,
             BaseMixedRealityProfile profile = null) : base()
         {
+            if (service == null)
+            {
+                Debug.LogError($"{name} requires a valid input system instance.");
```

Comment:
error comment seems incorrect. BaseDataProvider can apply to other systems than Input system